pkg/common/utils/utils.go

```diff
@@ -104,12 +104,21 @@ func IsAssignedPod(pod *v1.Pod) bool {
 }
 
 func GetQueueNameFromPod(pod *v1.Pod) string {
+	// Queue name can be defined in multiple places
+	// The queue name is determined by the following order
+	// 1. Label: constants.CanonicalLabelQueueName
+	// 2. Label: constants.LabelQueueName
+	// 3. Annotation: constants.AnnotationQueueName
+	// 4. Default: constants.ApplicationDefaultQueue
 	queueName := constants.ApplicationDefaultQueue
-	if an := GetPodLabelValue(pod, constants.LabelQueueName); an != "" {
-		queueName = an
-	} else if qu := GetPodAnnotationValue(pod, constants.AnnotationQueueName); qu != "" {
-		queueName = qu
+	if canonicalLabelQueueName := GetPodLabelValue(pod, constants.CanonicalLabelQueueName); canonicalLabelQueueName != "" {
+		queueName = canonicalLabelQueueName
+	} else if labelQueueName := GetPodLabelValue(pod, constants.LabelQueueName); labelQueueName != "" {
+		queueName = labelQueueName
+	} else if annotationQueueName := GetPodAnnotationValue(pod, constants.AnnotationQueueName); annotationQueueName != "" {
+		queueName = annotationQueueName
 	}
+
 	return queueName
```

Comment:
The order to get 'queue' from pod:

Before this PR:
1. Label: constants.LabelQueueName
2. Annotation: constants.AnnotationQueueName
3. Default: constants.ApplicationDefaultQueue

After this PR
1. **Label: constants.CanonicalLabelQueueName (New)**
2. Label: constants.LabelQueueName
3. Annotation: constants.AnnotationQueueName
4. Default: constants.ApplicationDefaultQueue